webapp/src/pages/ChatRoom/ChatRoom.tsx

```diff
@@ -29,7 +29,7 @@ function ChatRoom() {
   return (
     <div className="chat-room">
       <div className="chat-room-header">
-        <Header icon={IconVariant.ARROW_BACK} onClick={goBack}>
+        <Header icon={IconVariant.ARROW_BACK} onClick={goBack()}>
```

Comment:
suggestion (if-minor): To prevent this bug in the future, we could change the `goBack` function to something like this:

```diff
diff --git a/webapp/src/shared/hooks/UseNavigation.tsx b/webapp/src/shared/hooks/UseNavigation.tsx
index 16dcc94..a1bd92a 100644
--- a/webapp/src/shared/hooks/UseNavigation.tsx
+++ b/webapp/src/shared/hooks/UseNavigation.tsx
@@ -1,20 +1,14 @@
 import { NavigateFunction, useNavigate } from 'react-router-dom';
 
-type Callback = () => void;
-
 type UseNavigationReturn = {
   navigate: NavigateFunction;
-  goBack: () => Callback;
+  goBack: VoidFunction;
 };
 
 export const useNavigation = (): UseNavigationReturn => {
   const navigate = useNavigate();
 
-  const goBack = () => {
-    return () => {
-      navigate(-1);
-    };
-  };
+  const goBack = () => navigate(-1);
 
   return { navigate, goBack };
 };
```

Reply:
why not just use the hook we already have?https://github.com/42AGV/ft_transcendence/blob/7befac8cae299e7e84a97eb3573cc4619770b22c/webapp/src/shared/hooks/UseNavigation.tsx#L13

Reply:
We already are using that hook. You mean why don't we use it in its current form?